core/src/test/java/com/linecorp/armeria/server/http/auth/AuthServiceTest.java

```diff
@@ -0,0 +1,108 @@
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponseWriter;
+import com.linecorp.armeria.common.http.HttpStatus;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.http.AbstractHttpService;
+import com.linecorp.armeria.server.logging.LoggingService;
+import com.linecorp.armeria.test.AbstractServerTest;
+
+public class AuthServiceTest extends AbstractServerTest {
+
+    @Override
+    protected void configureServer(ServerBuilder sb) throws Exception {
+        final Map<String, String> usernameToPassword = ImmutableMap.of("brown", "sally", "pangyo", "choco");
+        Predicate<BasicToken> predicate = new Predicate<BasicToken>() {
+            @Override
+            public boolean test(BasicToken basicToken) {
+                String username = basicToken.username();
+                String password = basicToken.password();
+                return password.equals(usernameToPassword.get(username));
+            }
+        };
+
+        sb.serviceAt(
+                "/",
+                new AbstractHttpService() {
+                    @Override
+                    protected void doGet(
+                            ServiceRequestContext ctx, HttpRequest req, HttpResponseWriter res) {
+                        res.respond(HttpStatus.OK);
+                    }
+                }.decorate(HttpBasicAuthService.newDecorator(predicate))
+                 .decorate(LoggingService::new));
+    }
+
+    @Test
+    public void testBasicAuth() throws Exception {
+        try (CloseableHttpClient hc = HttpClients.createMinimal()) {
+            try (CloseableHttpResponse res = hc.execute(
+                    basicGetRequest("/", BasicToken.of("brown", "sally")))) {
+                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 200 OK"));
+            }
+            try (CloseableHttpResponse res = hc.execute(
+                    basicGetRequest("/", BasicToken.of("pangyo", "choco")))) {
+                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 200 OK"));
+            }
+            try (CloseableHttpResponse res = hc.execute(
+                    basicGetRequest("/", BasicToken.of("choco", "pangyo")))) {
+                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 401 Unauthorized"));
+            }
+            try (CloseableHttpResponse res = hc.execute(new HttpGet(uri("/")))) {
+                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 401 Unauthorized"));
+            }
+            Map<String, String> oAuth1aParams = Maps.newHashMap();
```

Comment:
I think it is better to use `ImmutableMap.<String, String>builder().put(..) ... .build()`